power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/iterative_linear_se_solver.hpp

```diff
@@ -681,9 +749,11 @@ template <bool sym> class IterativeLinearSESolver {
                     // shunt
                     if (type == YBusElementType::shunt) {
                         if (measured_value.has_shunt(obj)) {
-                            // G += Ys^H * Ys / variance
-                            block.g() += dot(hermitian_transpose(param.shunt_param[obj]), param.shunt_param[obj]) /
-                                         measured_value.shunt_power(obj).variance;
+                            // G += Ys^H * Ys * (variance^-1)
+                            auto const& shunt_power = measured_value.shunt_power(obj);
+                            block.g() += dot(hermitian_transpose(param.shunt_param[obj]), param.shunt_param[obj],
+                                             inv(ComplexTensor<sym>{static_cast<ComplexValue<sym>>(
```

Comment:
Do not use inverse. This is a known diagonal matrix. 

(
Comment
Do not use inverse. This is a known diagonal matrix. Use something like:

```
to_diag(1.0 / (shunt_power.p_variance + shunt_power.q_variance))
```